### PR TITLE
Public APIs to generate custom WKWebViewConfiguration with PKeyAuth UserAgent Str

### PR DESCRIPTION
**Pull request by NerevarineRule**
## Proposed changes

Added public APIs on iOS & MacOS to generate custom WKWebViewConfiguration with PKeyAuth UserAgent Keyword.

## Type of change

- [ ] Feature work
- [x] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [x] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [ ] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: IdentityCore/src/webview/embeddedWebview/ui/ios/MSIDWebviewUIController.m [hunk 1]
```diff
@@ -60,6 +60,14 @@ + (void)initialize
     });
 }
 
++ (WKWebViewConfiguration *)createWebViewConfigWithPKeyAuthUserAgent
+{
+    WKWebViewConfiguration *webConfig = [WKWebViewConfiguration new];
```
Comment (oldalton) [inline IdentityCore/src/webview/embeddedWebview/ui/ios/MSIDWebviewUIController.m:65]: This also needs to include preferred content mode for iOS, and have a corresponding public API in ADAL space.

File: IdentityCore/src/webview/embeddedWebview/ui/mac/MSIDWebviewUIController.m [hunk 2]
```diff
@@ -52,6 +54,13 @@ + (void)initialize
     });
 }
 
++ (WKWebViewConfiguration *)createWebViewConfigWithPKeyAuthUserAgent
+{
+    WKWebViewConfiguration *webConfig = [WKWebViewConfiguration new];
+    webConfig.applicationNameForUserAgent = kMSIDPKeyAuthKeyWordForUserAgent;
```
Comment (oldalton) [inline IdentityCore/src/webview/embeddedWebview/ui/mac/MSIDWebviewUIController.m:60]: I don't think we support PkeyAuth on macOS for ADAL. Let's remove it from this version.

File: IdentityCore/src/webview/embeddedWebview/ui/mac/MSIDWebviewUIController.m [hunk 3]
```diff
@@ -44,6 +45,7 @@ + (void)initialize
     static dispatch_once_t onceToken;
     dispatch_once(&onceToken, ^{
         s_webConfig = [WKWebViewConfiguration new];
+        s_webConfig.applicationNameForUserAgent = kMSIDPKeyAuthKeyWordForUserAgent;
```
Comment (oldalton) [inline IdentityCore/src/webview/embeddedWebview/ui/mac/MSIDWebviewUIController.m:48]: We don't support PkeyAuth on macOS for ADAL.

**Review by oldalton — CHANGES_REQUESTED** [review]
Added some comments.

File: IdentityCore/src/webview/embeddedWebview/ui/MSIDWebviewUIController.h [hunk 4]
```diff
@@ -40,6 +41,9 @@ NSWindowController
 #if TARGET_OS_IPHONE
 @property (weak) UIViewController *parentController;
 @property UIModalPresentationStyle presentationType;
+
++ (WKWebViewConfiguration *)defaultWKWebviewConfiguration;
```
Comment (kaisong1990) [inline IdentityCore/src/webview/embeddedWebview/ui/MSIDWebviewUIController.h:45]: Any reason not declare it as a property?

Comment (NerevarineRule) [reply IdentityCore/src/webview/embeddedWebview/ui/MSIDWebviewUIController.h:45]: That way we don't have to initialize the webview controller when calling it in the ADAL public API :)

Comment (oldalton) [reply IdentityCore/src/webview/embeddedWebview/ui/MSIDWebviewUIController.h:45]: Maybe Kai meant a class property :)?

**Review by oldalton — APPROVED** [review]

File: IdentityCore/src/webview/embeddedWebview/ui/MSIDWebviewUIController.h [hunk 5]
```diff
@@ -25,6 +25,7 @@
 
 #import <Foundation/Foundation.h>
 #import <WebKit/WebKit.h>
+#import "MSIDWorkPlaceJoinConstants.h"
```
Comment (oldalton) [inline IdentityCore/src/webview/embeddedWebview/ui/MSIDWebviewUIController.h:28]: is this needed here?

**Review by oldalton — APPROVED** [review]

File: IdentityCore/src/webview/embeddedWebview/ui/MSIDWebviewUIController.h (hunk 5 above)
Comment (NerevarineRule) [reply IdentityCore/src/webview/embeddedWebview/ui/MSIDWebviewUIController.h:28]: Technically, we can put in in the MSIDWebviewUIController.m file, but this is a project target header, so I don't see any major concerns. It is needed in the implementation file to recognize the PKeyAuth constant variable.

Comment (oldalton) [reply IdentityCore/src/webview/embeddedWebview/ui/MSIDWebviewUIController.h:28]: Yes, let's move it to .m file :)

**Comment by oldalton** [comment]
Check Changelogs CI check not applicable here, since this legacy ADAL fork. Overriding merge PR.

**Pull request merged**